src/server/api/controllers/orders.controller.js

```diff
@@ -5,50 +5,67 @@ const getOrders = async () => {
   return knex('orders');
 };
 
-const getOrderByUserId = async (userid) => {
-  if (!userid) {
+const getOrderById = async (id) => {
+  if (!id) {
     throw new HttpError(
-      'Bad request.  userid must be an integer and larger than 0',
+      'Bad request. Order ID must be an integer and larger than 0',
       400,
     );
   }
+
   try {
-    const orders = await knex('orders').where({ user_id: userid });
+    const orders = await knex('orders AS o')
+      .select(
+        'o.id as orderId',
+        'o.status as orderStatus',
+        'o.created_at as orderDate',
+        'o.user_id',
+        'oi.quantity',
+        'p.id as productId',
+        'p.name',
+        'p.price',
+        'p.color',
+        'p.size',
+        'p.picture',
+        'p.stock_amount',
+        'p.price',
+      )
+      .join('order_items AS oi ', 'o.id', '=', 'oi.order_id')
+      .join('products AS p', 'p.id', '=', 'oi.product_id')
+      .where('o.id', '=', id);
+
     if (orders.length === 0) {
-      throw new Error(` order with the specified userid was not found`, 404);
+      throw new Error(
+        `A order with the specified ID was not found: ${id}`,
+        404,
+      );
     }
     return orders;
   } catch (error) {
     return error.message;
   }
 };
-const getOrdersById = async (id) => {
-  if (!id) {
+
+const getOrderByUserId = async (userid) => {
+  if (!userid) {
     throw new HttpError(
-      'Bad request. Order ID must be an integer and larger than 0',
+      'Bad request.  userid must be an integer and larger than 0',
       400,
     );
   }
-
   try {
-    const orders = await knex('orders').where({ id });
+    const orders = await knex('orders').where({ user_id: userid });
     if (orders.length === 0) {
-      throw new Error(
-        `An order with the specified ID was not found : ${id}`,
-        404,
-      );
+      throw new Error(` order with the specified userid was not found`, 404);
     }
     return orders;
   } catch (error) {
     return error.message;
   }
 };
-const storeNewOrder = async (data) => {
-  await knex('orders').insert(data);
-};
```

Comment:
Yes I removed post.

Reply:
Hey @zmzade. You have to add this parts in your PR. Because with this style if you merge that PR all post parts will delete from develop branch. So you need to add this 3 changes back your PR.